grpc-gateway/pb/service.proto

```diff
@@ -99,7 +99,8 @@ service GrpcGateway {
     };
   }
 
-  // Subscribe to events
+  // When the client creates a subscription, with include_current_state=true it can send duplicate events.
```

Comment:
It's nice to have it here, but it belongs to swagger. That's where users would like to be informed about this. I asked in a different comment, but why duplication occurs during subscription? Is it due to the race condition which occurs when you subscribe + retrieve from RD, while during retrieval from RD some event might be already received + eventually also processed and returned by the RD?

Reply:
yes correct

Reply:
I don't like that client has to do deduplication in the websocket. Why you cannot get values, merge them with current subscription diff and send after that?

Reply:
Otherwise, we need to hold version per resource even type: for 10000 devices with 40 resources, it is approx 400000 hashes which holds version per subscription. cca 30MB RAM per subscription so it's not cheap and we need to dropped in one time.

Reply:
I don't mean deduplication during the whole time subscription exists. This wouldn't be a good idea. Only initial request when include_current_state=true is set. It's the same as I mentioned in a previous comment.

Reply:
Do you know the time? I don't know it because processing events from NATS can take more than initialization.

Reply:
It’s the time of response from RD. Not from nats?

Reply:
It's not guaranteed :), the time from RD and processing events from NATS are not the same. 
If time from NATS queue processing <= RD processing then it's OK
But how do you want to guarantee it when the time for NATS queue processing > RD processing? - this can cause duplication

Reply:
Nats doesn’t confirm successful subscription?

Reply:
no - and it doesn't matter - confirm successful subscription is about lose events , for deduplication it doesn't helps